src/components/CharacterAssignment.jsx

```diff
@@ -16,85 +16,113 @@ const CharacterAssignment = ({ actors, currentScriptId }) => {
+   * @returns {object} newColumns: Object of actor column objects, actorIds: array of actor ids, represents column order
    */
-    // Create and update the columns for each of the actors and an Unassigned Characters column
-  const createOrUpdateActorColumns = (actors) => {
-      if (!currentScriptId && actors.length == 0) {
-        // Initial setup of actor columns, no script is selected
-        createInitialActorColumns(actors);
-        return;
+  const createInitialActorColumns = (actors) => {
+    const newColumns = {
+      'unassignedCharacters': {
+        id: 'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
       }
-
-      // Script is already set, only add or remove actors
-      const newColumns = { ...columns };
-      const newColumnList = [...columnList];
+    };
+    const actorIds = [];
+    actors.forEach((actor) => {
+      newColumns[actor.id] = {
+        id: actor.id.toString(),
+        title: actor.name,
+        characterIds: [],
+        lineCount: 0
+      };
+      actorIds.push(actor.id.toString());
+    });
+    return {newColumns, actorIds}
+  };
+  /**
+   * Update the actor columns when the number of actors changes and sets the new column values and order
+   * @param actors Array of objects with properties of name and id
+   */
+  const updateActorColumns = (actors) => {
+      const oldColumns = { ...columns };
+      const oldColumnOrder = [...columnOrder];
       // If there are more actors than columns, add a column
-      if (actors.length > newColumnList.length) {
-        addActorColumn(actors, newColumns, newColumnList);
+      if (actors.length > oldColumnOrder.length) {
+        const { newColumns, newColumnOrder} = addActorColumn(actors, oldColumns, oldColumnOrder);
+        setColumns(newColumns)
+        setColumnOrder(newColumnOrder)
       } else {
         // Remove an actor, there are more columns than actors
-        removeActorColumn(actors, newColumns, newColumnList);
+        const { newColumns, newColumnOrder } = removeActorColumn(actors, oldColumns, oldColumnOrder);
+        setColumns(newColumns)
+        setColumnOrder(newColumnOrder)
       }
+
     };
 
   /**
    * Adds a new actor column by checking that each actor in our list has a column,
    *  and creating one when a column isn't found with that actor's id
-   * @param actors - Array of objects with properties of name and id
-   * @param newColumns - Object containing column objects with properties of id, title, characterIds, and lineCount
-   * @param newColumnList - Array of column ids
+   * @param actors Array of objects with properties of name and id
+   * @param newColumns Object containing column objects with properties of id, title, characterIds, and lineCount
+   * @param newColumnOrder Array of column ids
    */
-  const addActorColumn = (actors, newColumns, newColumnList) => {
+  const addActorColumn = (actors, newColumns, newColumnOrder) => {
```

Comment:
nit: The "new" prefix probably isn't necessary here since this function doesn't need to know "new" from "old". It just adds the actor to the columns passed in whether those columns are new or not.  
  
Aside from that, the naming is also inconsistent with how it's being called elsewhere. Passing in `old` but the parameter is named `new`:
```js 
const { newColumns, newColumnOrder} = addActorColumn(actors, oldColumns, oldColumnOrder);
...
const addActorColumn = (actors, newColumns, newColumnOrder) => {
```